pkg/resource/model_package/sdk.go

```diff
@@ -937,13 +976,20 @@ func (rm *resourceManager) newUpdateRequestPayload(
 	if r.ko.Spec.ApprovalDescription != nil {
 		res.SetApprovalDescription(*r.ko.Spec.ApprovalDescription)
 	}
+	if r.ko.Spec.CustomerMetadataProperties != nil {
+		f1 := map[string]*string{}
+		for f1key, f1valiter := range r.ko.Spec.CustomerMetadataProperties {
+			var f1val string
+			f1val = *f1valiter
+			f1[f1key] = &f1val
+		}
+		res.SetCustomerMetadataProperties(f1)
+	}
 	if r.ko.Spec.ModelApprovalStatus != nil {
 		res.SetModelApprovalStatus(*r.ko.Spec.ModelApprovalStatus)
 	}
 	if r.ko.Status.ACKResourceMetadata != nil && r.ko.Status.ACKResourceMetadata.ARN != nil {
 		res.SetModelPackageArn(string(*r.ko.Status.ACKResourceMetadata.ARN))
-	} else {
-		res.SetModelPackageArn(rm.ARNFromName(*r.ko.Spec.ModelPackageName))
```

Comment:
Did we want to lose this?

Reply:
I believe this came as part of the automated codegeneration. Since this is in `sdkUpdate` and only a versioned ModelPackage can be updated I believe it is okay to lose this. Reason being that a `VersionedModelPackage` doesn't have a `ModelPackageName` and we set the `ModelPackageName` as the `ARN` in `sdkCreate` so this else statement is redundant.

Reply:
I'm happy if you're happy